wrangler-service/src/main/java/io/cdap/wrangler/service/adls/ADLSUtility.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.wrangler.service.adls;
+
+import com.microsoft.azure.datalake.store.ADLStoreClient;
+import com.microsoft.azure.datalake.store.DirectoryEntry;
+import com.microsoft.azure.datalake.store.DirectoryEntryType;
+import io.cdap.wrangler.proto.adls.ADLSDirectoryEntryInfo;
+import io.cdap.wrangler.proto.adls.FileQueryDetails;
+import io.cdap.wrangler.service.FileTypeDetector;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ADLSUtility {
+
+    private static final FileTypeDetector detector = new FileTypeDetector();
+
+    public static String testConnection(ADLStoreClient client) throws IOException {
+        try {
+            client.enumerateDirectory("/");
+        } catch (IOException e) {
+            throw new IOException("Connection Failed, please check given credentials : " + e.getMessage());
+        }
+        return "Success";
+    }
+
+    public static List<ADLSDirectoryEntryInfo> initClientReturnResponse(ADLStoreClient client, String path)
+            throws IOException {
+        if (!client.checkExists(path)) {
+            throw new IOException("Given path doesn't exist");
```

Comment:
This seems like it should result in a 404. Throwing an IOException will result in a 500, which is an internal error whereas this is a user error.